### PR TITLE
integration tests v2

### PR DESCRIPTION
**Pull request by riley-stride**
modify the integration tests to continue upon detecting log lines, rather than waiting predefined amounts of time. this way they move faster and are more deterministic.

### CODE REVIEW

File: x/stakeibc/keeper/icacallbacks_delegate.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ func DelegateCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, txM
 	}
 	k.Logger(ctx).Info(fmt.Sprintf("DelegateCallback %v", delegateCallback))
 	hostZone := delegateCallback.GetHostZoneId()
+	k.Logger(ctx).Info(fmt.Sprintf("[DELEGATION] success on %s", hostZone))
```
Comment (asalzmann) [inline x/stakeibc/keeper/icacallbacks_delegate.go:54]: nit: move to the end of the callback function

File: x/stakeibc/keeper/icacallbacks_redemption.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ func (k Keeper) UnmarshalRedemptionCallbackArgs(ctx sdk.Context, redemptionCallb
 }
 
 func RedemptionCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, txMsgData *sdk.TxMsgData, args []byte) error {
-	logMsg := fmt.Sprintf("RedemptionCallback executing packet: %d, source: %s %s, dest: %s %s",
+	logMsg := fmt.Sprintf("RedemptionCallback on executing packet: %d, source: %s %s, dest: %s %s",
```
Comment (asalzmann) [inline x/stakeibc/keeper/icacallbacks_redemption.go:34]:
```suggestion
	logMsg := fmt.Sprintf("RedemptionCallback executing on packet: %d, source: %s %s, dest: %s %s",
```

File: x/stakeibc/keeper/icacallbacks_redemption.go [hunk 3]
```diff
@@ -91,6 +91,7 @@ func RedemptionCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, t
 			}
 			userRedemptionRecord.IsClaimable = true
 			k.RecordsKeeper.SetUserRedemptionRecord(ctx, userRedemptionRecord)
+			k.Logger(ctx).Info(fmt.Sprintf("[REDEMPTION] completed on %s", hostZoneId))
```
Comment (asalzmann) [inline x/stakeibc/keeper/icacallbacks_redemption.go:94]: this loop is over `userRedemptionRecords`, I think we want this log line at the end of the function?

File: scripts-local/vars.sh [hunk 4]
```diff
@@ -239,7 +239,7 @@ WAIT_FOR_NONEMPTY_BLOCK () {
 }
 
 WAIT_FOR_STRING () {
-  ( tail -f -n0 $1 & ) | grep -q -E "$2"
```
Comment (asalzmann) [inline scripts-local/vars.sh:242]: what is this change?

**Review by asalzmann — COMMENTED** [review]
nice! can you add a description to the PR? should be good to merge after that

File: x/stakeibc/keeper/icacallbacks_delegate.go (hunk 1 above)
Comment (riley-stride) [reply x/stakeibc/keeper/icacallbacks_delegate.go:54]: done

File: x/stakeibc/keeper/icacallbacks_redemption.go (hunk 3 above)
Comment (riley-stride) [reply x/stakeibc/keeper/icacallbacks_redemption.go:94]: good catch. done

File: scripts-local/vars.sh (hunk 4 above)
Comment (riley-stride) [reply scripts-local/vars.sh:242]: remove the regex `-E` option on grep

**Review by asalzmann — APPROVED** [review]

**Review by asalzmann — COMMENTED** [review]
actually I could not get these to run locally - will try again

**Pull request merged**